### ISSUE TITLE
[BUG] GUI diameter is reset to 30 when training own model

### ISSUE THREAD

**Issue by manerotoni** [MouseLand/cellpose#963]
**Describe the bug**
Using cellpose 3.0.7 
Training a model with image and masks from directory resets the diameter to default 30. As in my case the diameter is 80, this leads to a poor training. 
 

**To Reproduce**
1. Go to a directory that contains images and segmention masks. Load one image
2. Change diameter to something different than 30
3. [Models > Train model with image and masks in folder]
4. Look at the prompt it will show ">>>> model diam_mean =  30.000 (ROIs rescaled to this size during training)"
5. The diameter returned to 30 and the classifier has also quite a bad performance

**Comment by carsen-stringer** [comment]
This is fixed in the latest release, I'll reopen if it's not working for you

**Comment by manerotoni** [comment]
Thanks for the quick reply. I upgrade cellpose3 and the diameter is now not reverted to 30.

However, I am not sure if there is still not a problem. I checked cellpose 2 and 3 side by side and I consistently get worse results for cellpose3. Although the starting models are different,  given enough training data one should get a similar result. 

The loss function in cellpose2 reaches values <0.2 (0.16), in cellpose 3 i barely get close to 0.4 (typically >0.4). The segmentation results in cellpose3 are significantly worse. I tried several different models and the results are consistent. I will rescale the data for cellpose3 and see if I get a different result.

## train model in cellpose2
Cellpose2 gives me an hint that the the mean of training label mask is not 30 

```
>>>> mean of training label mask diameters (saved to model) 68.676
```
## train model in cellpos3
I do not get an hint whether the labels have been rescaled or not.

```
2024-06-18 16:09:49,163 [INFO] >>>> model diam_mean =  30.000 (ROIs rescaled to this size during training)
GUI_INFO: name of new model: cyto_20240618_155600
2024-06-18 16:09:49,690 [INFO] computing flows for labels
100%|███████████████████████████████████████████████████████████████████████████████████████████████| 16/16 [00:02<00:00,  6.20it/s]
2024-06-18 16:09:52,616 [INFO] >>>> median diameter set to = 30
2024-06-18 16:09:52,617 [INFO] >>>> mean of training label mask diameters (saved to model) 68.676
``` 
2024-06-18 16:11:10,542 [INFO] >>>> model diam_mean =  30.000 (ROIs rescaled to this size during training)
GUI_INFO: name of new model: cyto3_cp3_20240618_155847
2024-06-18 16:11:10,542 [INFO] computing flows for labels
100%|██████████████████████████████████████████████████████████████████████████████████| 16/16 [00:02<00:00,  7.35it/s]
2024-06-18 16:11:12,727 [INFO] >>> computing diameters
100%|█████████████████████████████████████████████████████████████████████████████████| 16/16 [00:00<00:00, 422.34it/s]
2024-06-18 16:11:12,765 [INFO] >>> using channels [0, 0]
```

**Comment by carsen-stringer** [comment]
oh there was another bug with the nimg_per_epoch that I just fixed, but I'm
not sure if that's enough to make such a large difference. please try pip
install git+https://github.com/mouseland/cellpose.git

**Comment by manerotoni** [comment]
Dear @carsen-stringer,
I changed the title of the bug as it is probably not related to the diameter size. I resized the data so that the objects have a diameter around 30. 

I consistently get worse results with cellpose3 than cellpose2.

Cellpose 3 I use 
```
2024-07-01 10:21:44,170 [INFO] WRITING LOG OUTPUT TO C:\Users\apoliti\.cellpose\run.log
2024-07-01 10:21:44,170 [INFO]
cellpose version:       3.0.11.dev1+g84344b0
platform:               win32
python version:         3.10.14
torch version:          2.3.1
2024-07-01 10:21:44,576 [INFO] ** TORCH CUDA version installed and working. **
```

cellpose 2 I use  

```
2024-07-01 10:26:52,634 [INFO] WRITING LOG OUTPUT TO C:\Users\apoliti\.cellpose\run.log
2024-07-01 10:26:52,634 [INFO]
cellpose version:       2.3.2
platform:               win32
python version:         3.10.14
torch version:          2.3.1
2024-07-01 10:26:52,828 [INFO] ** TORCH CUDA version installed and working. **
```

training is via the GUI using following data set [dataset_compare_cellpose2_3.zip](https://github.com/user-attachments/files/16050117/dataset_compare_cellpose2_3.zip)

**Comment by manerotoni** [comment]
## Training cellpose2 

start with TN1. With other models the results are similar.  

GT 
![image](https://github.com/MouseLand/cellpose/assets/3718770/c89f5505-03a6-4c0f-a7b6-5823670f8b06)

training metric, loss ~ 0.15
```
2024-07-01 10:34:22,810 [INFO] Epoch 5, Time  2.7s, Loss 1.0513, LR 0.0556
2024-07-01 10:34:24,887 [INFO] Epoch 10, Time  4.8s, Loss 0.6487, LR 0.1000
2024-07-01 10:34:29,042 [INFO] Epoch 20, Time  9.0s, Loss 0.3015, LR 0.1000
2024-07-01 10:34:33,214 [INFO] Epoch 30, Time 13.1s, Loss 0.2173, LR 0.1000
2024-07-01 10:34:37,399 [INFO] Epoch 40, Time 17.3s, Loss 0.1937, LR 0.1000
2024-07-01 10:34:41,522 [INFO] Epoch 50, Time 21.4s, Loss 0.1723, LR 0.1000
2024-07-01 10:34:45,690 [INFO] Epoch 60, Time 25.6s, Loss 0.1634, LR 0.1000
2024-07-01 10:34:49,790 [INFO] Epoch 70, Time 29.7s, Loss 0.1588, LR 0.1000
2024-07-01 10:34:53,911 [INFO] Epoch 80, Time 33.8s, Loss 0.1568, LR 0.1000
2024-07-01 10:34:57,990 [INFO] Epoch 90, Time 37.9s, Loss 0.1507, LR 0.1000
```
predicted masks look good
![image](https://github.com/MouseLand/cellpose/assets/3718770/9695a1eb-16a4-490f-86e8-e07d3efde298)

**Comment by manerotoni** [comment]
Honestly, what I see is probably data specific and also due to different starting models for cellpose2 and 3. 

I have now tried all models in cellpose 3 and I do find models that perform well (e.g. nuclei). Although the initial performance was much worse (bad predictions) and I would normally do not pick such model, the final result after training was very good.
